actioncable/app/javascript/action_cable/consumer.js

```diff
@@ -28,8 +28,9 @@ import Subscriptions from "./subscriptions"
 // automatically resubscribe.
 
 export default class Consumer {
-  constructor(url) {
+  constructor(url, token = null) {
```

Comment:
I would leave the existing constructor as it is and, instead, add a new method to add subprotocols. It can be as simple as:

```js
consumer = createConsumer("https://ws.example.com/cable")
consumer.addSubProtocol("some subprotocol")
```

Then, the consumer can expose a list of `consumer.subProtocols` to use when initializing the WebSocket connection.

I'd use the name *sub protocol* instead of *token* to make it clear what this is doing. Using the sub protocol as a way to transport a JWT token is what motivates this, but we are adding a broader functionality here.

Reply:
@zedtux also, we should document the change in `actioncable/CHANGELOG.md` (you can see the format in other frameworks such as `activerecord/CHANGELOG.md`, since right now it's empty for Action Cable).

Reply:
A quick update : I've already started to update my code, but I have to finish a milestone first. I'll come back on this on asap.

Reply:
I'm done with the milestone, I will be able to work on this very soon !

Reply:
I'm on it !

Reply:
@jorgemanrubia I'm done. I have spent quite some time looking to assert the HTTP header content in order to check the subprotocols sent by WebSocket but it [doesn't seem to be supported by mock-socket](https://github.com/thoov/mock-socket/blob/2.0.0/src/websocket.js#L33).

Let me know please if you see additional test to be added.